custom_components/tuya_local/devices/AlecoAir_D25_Traditio.yaml

```diff
@@ -0,0 +1,226 @@
+      name: speed
+      icon: "mdi:fan"
+      mapping:
+        - dps_val: low
+          value: low
+        - dps_val: high
+          value: high        
+    - id: 5
+      name: mode
+      type: string
+      mapping:
+        - dps_val: Manual
+          value: Manual
+        - dps_val: Auto
+          value: Auto
+          icon: "mdi:tshirt-crew-outline"
+          icon_priority: 3
+        - dps_val: Laundrydrying
+          value: Laundrydrying
+          icon: "mdi:air-filter"
+          icon_priority: 3
+        - dps_val: Sleep
+          value: Sleep
+          icon: "mdi:power-sleep"
+          icon_priority: 3
+        - dps_val: Purifying
+          value: Purifying
+          icon: "mdi:leaf"
+    - id: 19
+      name: error
+      type: bitfield
+      # Docs mention E1 and E2, it isn't clear which is tank full, so for now
+      # treat both as that.
+      mapping:
+        - dps_val: 0
+          value: OK
+        - dps_val: 1
+          value: E1
+          icon: "mdi:cup-water"
+          icon_priority: 2
+        - dps_val: 2
+          value: E2
+          icon: "mdi:cup-water"
+          icon_priority: 2
+    - id: 8
+      type: boolean
+      name: swing
+      mapping:
+        - dps_val: true
+          icon: "mdi:creation"
+        - dps_val: false
+          icon: "mdi:creation-outline"
+secondary_entities:
+  - entity: select
+    name: speed
+    icon: "mdi:fan"
+    category: config
+    dps:
+      - id: 4
+        type: string
+        name: option
+        icon: "mdi:fan"
+        mapping:
+          - dps_val: low
+            value: low
+          - dps_val: high
+            value: high
+  - entity: switch
+    name: swing
+    category: config
+    icon: "mdi:tailwind"
+    dps: 
+      - id: 8
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+            icon: "mdi:tailwind"
+          - dps_val: false
+            icon: "mdi:tailwind"
+  - entity: switch
+    name: ionizer
+    category: config
+    icon: "mdi:creation"
+    dps:
+      - id: 10
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+            icon: "mdi:creation"
+          - dps_val: false
+            icon: "mdi:creation-outline"
+  - entity: switch
+    name: uv
+    category: config
+    icon: "mdi:leaf"
+    dps:
+      - id: 13
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+            icon: "mdi:leaf"
+          - dps_val: false
+            icon: "mdi:leaf-outline"
+  - entity: sensor
+    name: Current humidity
+    class: humidity
+    dps:
+      - id: 6
+        type: integer
+        name: sensor
+        unit: "%"
+        class: measurement
+  - entity: binary_sensor
+    name: Tank
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 19
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: false
+          - value: true
```

Comment:
If you know which fault code is "tank full" (usually combined with tank missing), then it is better to do this the other way around - default value is false, and only the code for "tank full" is true.  Another general problem binary_sensor can be added to catch other errors (but should set the tank full code to false as well as the 0 code)